src/vm/instruction.rs

```diff
@@ -1,14 +1,9 @@
 use num_bigint::BigUint;
 
-num Register {
-    AP,
-    FP
-}
-
 pub struct Instruction {
-    off0: BigUint,
-    off1: BigUint,
-    off2: BigUint,
+    off0: i16,
```

Comment:
out of curiosity why are you changing the type @Juan-M-V ?

Reply:
Because the specification divides the 64-bit words into off0, off1, off2 and flags, each taking 16-bits

Reply:
Cool, thanks